Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1085 @@
+            auto const &flag_ext_face = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+            reduce_ops.eval(box, reduce_data,
+                [=] AMREX_GPU_DEVICE(int i, int j, int k) -> amrex::GpuTuple<int> {
+                    return flag_ext_face(i, j, k);
+                });
+        }
+
+        auto r = reduce_data.value();
+        sums(idim) = amrex::get<0>(r);
+    }
+
+    amrex::ParallelDescriptor::ReduceIntSum(&(sums(0)), AMREX_SPACEDIM);
+#endif
+    return sums;
+}
+
+
+void
+WarpX::ComputeFaceExtensions(){
+#ifdef AMREX_USE_EB
+    amrex::Array1D<int, 0, 2> N_ext_faces = CountExtFaces();
+    amrex::ParallelDescriptor::ReduceIntSum(&(N_ext_faces(0)), 3);
+    amrex::Print()<< "Faces to be extended in x:\t" << N_ext_faces(0) <<std::endl;
+    amrex::Print()<< "Faces to be extended in y:\t" << N_ext_faces(1) <<std::endl;
+    amrex::Print()<< "Faces to be extended in z:\t" << N_ext_faces(2) <<std::endl;
+
+    InitBorrowing();
+    amrex::Array1D<int, 0, 2> temp_inds = ComputeOneWayExtensions();
+    amrex::Array1D<int, 0, 2> N_ext_faces_after_one_way = CountExtFaces();
+    amrex::ParallelDescriptor::ReduceIntSum(&(N_ext_faces_after_one_way(0)), 3,
+                                            amrex::ParallelDescriptor::IOProcessorNumber());
+    amrex::Print()<< "Faces to be extended after one way extension in x:\t" <<
+                     N_ext_faces_after_one_way(0) <<std::endl;
+    amrex::Print()<< "Faces to be extended after one way extension in y:\t" <<
+                     N_ext_faces_after_one_way(1) <<std::endl;
+    amrex::Print()<< "Faces to be extended after one way extension in z:\t" <<
+                     N_ext_faces_after_one_way(2) <<std::endl;
+    ComputeEightWaysExtensions(temp_inds);
+    amrex::Array1D<int, 0, 2> N_ext_faces_after_eight_ways = CountExtFaces();
+    amrex::ParallelDescriptor::ReduceIntSum(&(N_ext_faces_after_eight_ways(0)), 3,
+                                            amrex::ParallelDescriptor::IOProcessorNumber());
+    amrex::Print()<< "Faces to be extended after eight ways extension in x:\t" <<
+                     N_ext_faces_after_eight_ways(0) <<std::endl;
+    amrex::Print()<< "Faces to be extended after eight ways extension in y:\t" <<
+                     N_ext_faces_after_eight_ways(1) <<std::endl;
+    amrex::Print()<< "Faces to be extended after eight ways extension ins z:\t" <<
+                     N_ext_faces_after_eight_ways(2) <<std::endl;
+
+    if (N_ext_faces_after_eight_ways(0) > 0) {
+        amrex::Abort("Some x faces could not be extended");
+    }
+    if (N_ext_faces_after_eight_ways(1) > 0) {
+        amrex::Abort("Some y faces could not be extended");
+    }
+    if (N_ext_faces_after_eight_ways(2) > 0) {
+        amrex::Abort("Some z faces could not be extended");
+    }
+#endif
+}
+
+
+void
+WarpX::InitBorrowing() {
+    int idim = 0;
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][0]); mfi.isValid(); ++mfi) {
```

Comment:
Small change for consistency
```suggestion
    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
```